x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+        expect(CredentialsLogic.values.activeApiToken).toEqual({
+          ...newToken,
+          foo: true,
+        });
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setTokenReadWrite({ name: 'foo', checked: true });
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+      });
+    });
+  });
+
+  describe('setTokenName', () => {
+    describe('activeApiToken', () => {
+      it('update the name property on the activeApiToken, formatted correctly', () => {
+        mount({
+          activeApiToken: {
+            ...newToken,
+            name: 'bar',
+          },
+        });
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.activeApiToken).toEqual({
+          ...newToken,
+          name: 'new-name',
+        });
+      });
+    });
+
+    describe('activeApiTokenRawName', () => {
+      it('updates the raw name, with no formatting applied', () => {
+        mount();
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual('New Name');
+      });
+    });
+
+    describe('keyHelpText', () => {
+      it('should update the keyHelpText value to reflect the new name', () => {
+        mount();
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.keyHelpText).toEqual('Your key will be named: new-name');
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setTokenName('New Name');
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+        activeApiTokenRawName: expect.any(String),
+        keyHelpText: expect.any(String),
+      });
+    });
+  });
+
+  describe('setTokenType', () => {
+    describe('activeApiToken.access_all_engines', () => {
+      describe('when value is ADMIN', () => {
+        it('updates access_all_engines to false', () => {
+          mount({
+            activeApiToken: {
+              ...newToken,
+              access_all_engines: true,
+            },
+          });
+          CredentialsLogic.actions.setTokenType(ADMIN);
+          expect(CredentialsLogic.values.activeApiToken.access_all_engines).toEqual(false);
+        });
+      });
+      describe('when value is not ADMIN', () => {
+        it('will maintain access_all_engines value when true', () => {
+          mount({
+            activeApiToken: {
+              ...newToken,
+              access_all_engines: true,
+            },
+          });
+          CredentialsLogic.actions.setTokenType(PRIVATE);
+          expect(CredentialsLogic.values.activeApiToken.access_all_engines).toEqual(true);
+        });
+        it('will maintain access_all_engines value when false', () => {
```

Comment:
lint nit - newline
```suggestion
        });

        it('will maintain access_all_engines value when false', () => {
```

Reply:
actually just to check - are your newlines / groupings intentional? if so, disregard these comments 🤷

Reply:
I audited and cleaned up the whitespace in this file quick: 1fa29d58508631a9c5111f73288937a646e105fb

Reply:
Awesome, thanks!!